src/backend/instagram/instagramConnection.ts

```diff
@@ -8,29 +8,25 @@ import { instagramEndpoints } from './instagramEndpoints';
 
 export const instagramConnectionState = trackConnectionState(10 * 60 * 1000);
 
-const isInvalidCode = (body: string) => {
-  return JSON.parse(body).error_message === 'Invalid authorization code';
+const isInvalidCode = (error: HTTPError) => {
+  const errorMessage = JSON.parse(error.response.body as string).error_message;
+  return errorMessage === 'Invalid authorization code';
```

Comment:
Here’s what I had in mind. This way you will not need the casting `as HTTPError` below. Please also note that we declare functions as function and not as constants.
```suggestion
function isInvalidCode(error: unknown) {
  return (
    error instanceof HTTPError &&
    JSON.parse(error.response.body).error_message === 'Invalid authorization code'
  );
```